### ISSUE TITLE
Nested struct field's pattern matching

### ISSUE THREAD

**Issue by ratijas** [intellij-rust/intellij-rust#7037]
## Environment

* **IntelliJ Rust plugin version:** 0.3.143.3737-203
* **Rust toolchain version:** 1.50.0 (cb75ad5db 2021-02-10) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2020.3.3 (CL-203.7717.62)
* **Operating system:** Linux 5.11.10-arch1-1
* **Macro expansion engine:** new
* **Name resolution engine:** old

## Problem description

Pattern matching on nested struct's field is recognized by IntelliJ Rust as a binding, considers it an unused variable, and suggests renaming `None` to `_None`. But cargo/rustc does not warn about unused variables, and in runtime my program behaves just fine.

![image](https://user-images.githubusercontent.com/6737986/113021516-54dab380-918c-11eb-8144-5b35a9e33258.png)

In the meantime, matching & binding to `group: Some(group)` is recognized alright.

## Steps to reproduce

Consider the following structures
```rust
pub enum Command {
    Status(StatusCommand),
    // ...
}
pub struct StatusCommand {
    pub board: u8,
    pub group: Option<Group>,
}
```

And match against them somewhere in function body

```rust
match command {
    Command::Status(StatusCommand { board, group: None }) => {}
    Command::Status(StatusCommand { board, group: Some(group) }) => {}
    // ...
}
```

**Comment by ratijas** [comment]
Actually, just encountered this bug in a normal pattern matching without nested destructuring. This happened in a scratch file, so naturally semantic analysis is quite limited. Still, I didn't expect this to happen at all.


![`None` variant highlights as an unused binding](https://user-images.githubusercontent.com/6737986/113108165-5bf5d600-920d-11eb-9f0c-ba3337f20a80.png)
![PsiViewer of a match tree](https://user-images.githubusercontent.com/6737986/113108075-484a6f80-920d-11eb-80bf-73acc7569909.png)
